Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/ContainerActionButtonHandler.cpp

```diff
@@ -18,43 +18,72 @@ namespace AzToolsFramework
 
     void ContainerActionButtonHandler::SetValueFromDom(const AZ::Dom::Value& node)
     {
-        GenericButtonHandler::SetValueFromDom(node);
 
-        static QIcon s_iconAdd(QStringLiteral(":/stylesheet/img/UI20/add-16.svg"));
-        static QIcon s_iconRemove(QStringLiteral(":/stylesheet/img/UI20/delete-16.svg"));
-        static QIcon s_iconClear(QStringLiteral(":/stylesheet/img/UI20/delete-16.svg"));
-        static QIcon s_iconUp(QStringLiteral(":/stylesheet/img/indicator-arrow-up.svg"));
-        static QIcon s_iconDown(QStringLiteral(":/stylesheet/img/indicator-arrow-down.svg"));
+        static const QIcon s_iconAdd(QStringLiteral(":/stylesheet/img/UI20/add-16.svg"));
+        static const QIcon s_iconRemove(QStringLiteral(":/stylesheet/img/UI20/delete-16.svg"));
+        static const QIcon s_iconClear(QStringLiteral(":/stylesheet/img/UI20/delete-16.svg"));
+        static const QIcon s_iconUp(QStringLiteral(":/stylesheet/img/indicator-arrow-up.svg"));
+        static const QIcon s_iconDown(QStringLiteral(":/stylesheet/img/indicator-arrow-down.svg"));
 
         using AZ::DocumentPropertyEditor::Nodes::ContainerAction;
         using AZ::DocumentPropertyEditor::Nodes::ContainerActionButton;
 
+        GenericButtonHandler::SetValueFromDom(node);
+
+        auto oldAction = m_action;
         m_action = ContainerActionButton::Action.ExtractFromDomNode(node).value_or(ContainerAction::AddElement);
+
+        if (m_action == oldAction)
+        {
+            return;
+        }
+
         switch (m_action)
         {
         case ContainerAction::AddElement:
-            setIcon(s_iconAdd);
-            setToolTip("Add new child element");
-            break;
+            {
+                setIcon(s_iconAdd);
+                setToolTip("Add new child element");
```

Comment:
Should do a _tr here too
```suggestion
                setToolTip(tr("Add new child element"));
```